pkg/planner/cardinality/row_count_index.go

```diff
@@ -478,14 +478,29 @@ func expBackoffEstimation(sctx sessionctx.Context, idx *statistics.Index, coll *
 	})
 	if l == 1 {
 		return singleColumnEstResults[0], true, nil
-	} else if l == 2 {
-		return singleColumnEstResults[0] * math.Sqrt(singleColumnEstResults[1]), true, nil
-	} else if l == 3 {
-		return singleColumnEstResults[0] * math.Sqrt(singleColumnEstResults[1]) * math.Sqrt(math.Sqrt(singleColumnEstResults[2])), true, nil
 	} else if l == 0 {
 		return 0, false, nil
 	}
-	return singleColumnEstResults[0] * math.Sqrt(singleColumnEstResults[1]) * math.Sqrt(math.Sqrt(singleColumnEstResults[2])) * math.Sqrt(math.Sqrt(math.Sqrt(singleColumnEstResults[3]))), true, nil
+	// Do not allow the exponential backoff to go below the available index bound. If the number of predicates
+	// is less than the number of index columns - use 90% of the bound to differentiate a subset from full index match.
+	// If there is an individual column selectivity that goes below this bound, use that selectivity only.
+	idxLowBound := max(1/float64(idx.NDV), 1/float64(coll.RealtimeCount))
```

Comment:
```suggestion
	idxLowBound := 1/float64(min(idx.NDV, coll.RealtimeCount))
```

More simple.

Reply:
Are there situations where idx.NDV is unavailable? And will this result in us choosing an incorrect lower bound?

Reply:
Changed the code as recommended. But also added a check to make sure that we don't run into a divide by zero error for idx.NDV.